salt/beacons/__init__.py

```diff
@@ -214,6 +214,45 @@ def list_beacons(self):
 
         return True
 
+    def list_available_beacons(self):
+        '''
+        List the available beacons
+        '''
+        _beacons = ['{0}'.format(_beacon.replace('.beacon', ''))
+                    for _beacon in list(self.beacons) if '.beacon' in _beacon]
+
+        # Fire the complete event back along with the list of beacons
+        evt = salt.utils.event.get_event('minion', opts=self.opts)
+        evt.fire_event({'complete': True, 'beacons': _beacons},
+                       tag='/salt/minion/minion_beacons_list_available_complete')
+
+        return True
+
+    def validate_beacon(self, name, beacon_data):
+        '''
+        Return available beacon functions
+        '''
+        validate_str = '{}.validate'.format(name)
+        # Run the validate function if it's available,
+        # otherwise there is a warning about it being missing
+        if validate_str in self.beacons:
+            if 'enabled' in beacon_data:
+                del beacon_data['enabled']
+            valid, vcomment = self.beacons[validate_str](beacon_data)
+        else:
+            log.info('Beacon {0} does not have a validate'
+                     ' function,  skipping validation.'.format(name))
```

Comment:
One of the things I'm working on for Oxygen is doing away with using `str.format()` in logging. The `logging` module already performs printf-style string replacement using any of the `*args` passed to the log function, so this can be rewritten as:

```python
            log.info('Beacon %s does not have a validate function, '
                     'skipping validation.', name)
```